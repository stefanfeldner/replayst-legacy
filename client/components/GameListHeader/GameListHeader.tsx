import { useContext } from 'react';
import { View, StyleSheet } from 'react-native';
import { PALETTE } from '../../services/theme';
import { UserContext } from '../UserContext/UserContext';
import GameListHeaderButton from '../GameListHeaderButton/GameListHeaderButton';

export default function GameListHeader() {
  const { owned, favorites, wishlist } = useContext(UserContext);

  const [coll] = owned;
  const [favs] = favorites;
  const [wish] = wishlist;

  return (
    <View style={styles.container}>
      <GameListHeaderButton
        list={coll}
        listName={'owned'}
        iconed={'safe-square-outline'}
        icon={'safe-square'}
      />
      <GameListHeaderButton
        list={favs}
        listName={'favorites'}
        iconed={'heart-outline'}
        icon={'heart'}
      />
      <GameListHeaderButton
        list={wish}
        listName={'wishlist'}
        iconed={'star-outline'}
        icon={'star'}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginTop: 20
  },
  text: {
    color: PALETTE.one
  }
});
