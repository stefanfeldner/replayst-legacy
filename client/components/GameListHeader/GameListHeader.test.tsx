import GameListHeader from './GameListHeader';
import { render } from '@testing-library/react-native';

test('should display three buttons', () => {
  const tree = render(<GameListHeader />);
  expect(tree).toMatchSnapshot();
});
