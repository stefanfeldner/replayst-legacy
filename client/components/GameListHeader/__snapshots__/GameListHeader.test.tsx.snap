// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display three buttons 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-around",
      "marginTop": 20,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomLeftRadius": 10,
            "borderBottomRightRadius": 10,
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
            "height": 35,
            "justifyContent": "center",
            "width": 110,
          },
          Object {
            "backgroundColor": "#20150d",
          },
        ]
      }
    >
      <Text />
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomLeftRadius": 10,
            "borderBottomRightRadius": 10,
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
            "height": 35,
            "justifyContent": "center",
            "width": 110,
          },
          Object {
            "backgroundColor": "#20150d",
          },
        ]
      }
    >
      <Text />
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomLeftRadius": 10,
            "borderBottomRightRadius": 10,
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
            "height": 35,
            "justifyContent": "center",
            "width": 110,
          },
          Object {
            "backgroundColor": "#20150d",
          },
        ]
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
